docs/source/tutorial_torch_generator_agent.md

```diff
@@ -0,0 +1,154 @@
+# Using Torch Generator Agent
+
+**Authors**: Stephen Roller, Eric Smith
+
+`parlai.core.torch_generator_agent.TorchGeneratorAgent` is an abstract parent class that provides functionality for building autoregressive generative models. Extending `TorchGeneratorAgent` requires your model conform to a strict interface, but then provides you rich functionality like beam search and sampling.
+
+
+## Example Models
+
+Two major models in ParlAI inherit from `TorchGeneratorAgent`: seq2seq and transformer. You can try the transformer with the example below:
+
+```
+python examples/train_model -m transformer/generator -t convai2 -mf /tmp/testtransformer \
+  --beam-size 5 -bs 16
+```
+
+## Creating a Model
+
+In order to write a generative model, your agent should extend `parlai.core.torch_generator_agent.TorchGeneratorAgent`. This parent class implements `train_step` and `eval_step`, so you only need to implement your model and instantiate it through `build_model`. However, your model should implement the `parlai.core.torch_generator_agent.TorchGeneratorModel` interface.
+
+
+## Tutorial
+
+This tutorial will walk you through creating a simple generative model, found at `parlai.agents.examples.seq2seq`, that consists of an LSTM-based encoder and decoder.
+
+### Extending `TorchGeneratorAgent`
+
+A minimal `TorchGeneratorAgent` only needs to implement `build_model()`, but if you want to specify any command-line arguments, you'll need to add `add_cmdline_args()` as well. This method first adds flags for `TorchGeneratorAgent` and then adds a `--hidden-size` flag for the hidden dimension of the LSTMs.
```

Comment:
"This method" --> "the implementation below" ?

Reply:
also, of waht LSTMs? can you describe teh example model you're implementing first?

Reply:
Right, was a bit vague - added a little more detail on this to this section and the one above